Compile error with cef_binary_3.2171.1979_macosx64

I am trying to resolve issue #8 by upgrading to a more current version of CEF. I am not much of a C developer so could  you help me troubleshoot this error? Thanks, G

```
Scanning dependencies of target cef-bootstrap
[ 91%] Building CXX object plugins/obs-browser/CMakeFiles/cef-bootstrap.dir/cef-bootstrap/main.cpp.o
[ 91%] Building CXX object plugins/obs-browser/CMakeFiles/cef-bootstrap.dir/shared/browser-app.cpp.o
[ 92%] Linking CXX executable "CEF Helper.app/Contents/MacOS/CEF Helper"
Undefined symbols for architecture x86_64:
  "std::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> >::size() const", referenced from:
      cef::logging::operator<<(std::ostream&, wchar_t const*) in libcef_dll_wrapper.a(cef_logging.o)
  "std::string::c_str() const", referenced from:
      cef::logging::LogMessage::~LogMessage() in libcef_dll_wrapper.a(cef_logging.o)
  "std::basic_ostringstream<char, std::char_traits<char>, std::allocator<char> >::str() const", referenced from:
      cef::logging::LogMessage::~LogMessage() in libcef_dll_wrapper.a(cef_logging.o)
      std::string* cef::logging::MakeCheckOpString<int, int>(int const&, int const&, char const*) in libcef_dll_wrapper.a(cef_logging.o)
      std::string* cef::logging::MakeCheckOpString<unsigned long, unsigned long>(unsigned long const&, unsigned long const&, char const*) in libcef_dll_wrapper.a(cef_logging.o)
      std::string* cef::logging::MakeCheckOpString<unsigned long, unsigned int>(unsigned long const&, unsigned int const&, char const*) in libcef_dll_wrapper.a(cef_logging.o)
      std::string* cef::logging::MakeCheckOpString<unsigned int, unsigned long>(unsigned int const&, unsigned long const&, char const*) in libcef_dll_wrapper.a(cef_logging.o)
      std::string* cef::logging::MakeCheckOpString<std::string, std::string>(std::string const&, std::string const&, char const*) in libcef_dll_wrapper.a(cef_logging.o)
  "std::allocator<char>::allocator()", referenced from:
      cef::logging::(anonymous namespace)::safe_strerror(int) in libcef_dll_wrapper.a(cef_logging.o)
  "std::allocator<char>::~allocator()", referenced from:
      cef::logging::(anonymous namespace)::safe_strerror(int) in libcef_dll_wrapper.a(cef_logging.o)
  "std::allocator<wchar_t>::allocator()", referenced from:
      cef::logging::operator<<(std::ostream&, wchar_t const*) in libcef_dll_wrapper.a(cef_logging.o)
  "std::allocator<wchar_t>::~allocator()", referenced from:
      cef::logging::operator<<(std::ostream&, wchar_t const*) in libcef_dll_wrapper.a(cef_logging.o)
  "std::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> >::basic_string(wchar_t const*, std::allocator<wchar_t> const&)", referenced from:
      cef::logging::operator<<(std::ostream&, wchar_t const*) in libcef_dll_wrapper.a(cef_logging.o)
  "std::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> >::~basic_string()", referenced from:
      cef::logging::operator<<(std::ostream&, wchar_t const*) in libcef_dll_wrapper.a(cef_logging.o)
  "std::ostream::operator<<(std::ostream& (*)(std::ostream&))", referenced from:
      cef::logging::LogMessage::~LogMessage() in libcef_dll_wrapper.a(cef_logging.o)
  "std::ostream::operator<<(int)", referenced from:
      std::string* cef::logging::MakeCheckOpString<int, int>(int const&, int const&, char const*) in libcef_dll_wrapper.a(cef_logging.o)
  "std::ostream::operator<<(unsigned int)", referenced from:
      std::string* cef::logging::MakeCheckOpString<unsigned long, unsigned int>(unsigned long const&, unsigned int const&, char const*) in libcef_dll_wrapper.a(cef_logging.o)
      std::string* cef::logging::MakeCheckOpString<unsigned int, unsigned long>(unsigned int const&, unsigned long const&, char const*) in libcef_dll_wrapper.a(cef_logging.o)
  "std::ostream::operator<<(unsigned long)", referenced from:
      std::string* cef::logging::MakeCheckOpString<unsigned long, unsigned long>(unsigned long const&, unsigned long const&, char const*) in libcef_dll_wrapper.a(cef_logging.o)
      std::string* cef::logging::MakeCheckOpString<unsigned long, unsigned int>(unsigned long const&, unsigned int const&, char const*) in libcef_dll_wrapper.a(cef_logging.o)
      std::string* cef::logging::MakeCheckOpString<unsigned int, unsigned long>(unsigned int const&, unsigned long const&, char const*) in libcef_dll_wrapper.a(cef_logging.o)
  "std::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(char const*, std::allocator<char> const&)", referenced from:
      cef::logging::(anonymous namespace)::safe_strerror(int) in libcef_dll_wrapper.a(cef_logging.o)
  "std::basic_string<char, std::char_traits<char>, std::allocator<char> >::~basic_string()", referenced from:
      cef::logging::LogMessage::LogMessage(char const*, int, std::string*) in libcef_dll_wrapper.a(cef_logging.o)
      cef::logging::LogMessage::LogMessage(char const*, int, int, std::string*) in libcef_dll_wrapper.a(cef_logging.o)
      cef::logging::LogMessage::~LogMessage() in libcef_dll_wrapper.a(cef_logging.o)
      cef::logging::ErrnoLogMessage::~ErrnoLogMessage() in libcef_dll_wrapper.a(cef_logging.o)
  "std::basic_ostringstream<char, std::char_traits<char>, std::allocator<char> >::basic_ostringstream(std::_Ios_Openmode)", referenced from:
      cef::logging::LogMessage::LogMessage(char const*, int, int) in libcef_dll_wrapper.a(cef_logging.o)
      cef::logging::LogMessage::LogMessage(char const*, int, std::string*) in libcef_dll_wrapper.a(cef_logging.o)
      cef::logging::LogMessage::LogMessage(char const*, int, int, std::string*) in libcef_dll_wrapper.a(cef_logging.o)
      std::string* cef::logging::MakeCheckOpString<int, int>(int const&, int const&, char const*) in libcef_dll_wrapper.a(cef_logging.o)
      std::string* cef::logging::MakeCheckOpString<unsigned long, unsigned long>(unsigned long const&, unsigned long const&, char const*) in libcef_dll_wrapper.a(cef_logging.o)
      std::string* cef::logging::MakeCheckOpString<unsigned long, unsigned int>(unsigned long const&, unsigned int const&, char const*) in libcef_dll_wrapper.a(cef_logging.o)
      std::string* cef::logging::MakeCheckOpString<unsigned int, unsigned long>(unsigned int const&, unsigned long const&, char const*) in libcef_dll_wrapper.a(cef_logging.o)
      ...
  "std::basic_ostringstream<char, std::char_traits<char>, std::allocator<char> >::~basic_ostringstream()", referenced from:
      cef::logging::LogMessage::~LogMessage() in libcef_dll_wrapper.a(cef_logging.o)
      std::string* cef::logging::MakeCheckOpString<int, int>(int const&, int const&, char const*) in libcef_dll_wrapper.a(cef_logging.o)
      std::string* cef::logging::MakeCheckOpString<unsigned long, unsigned long>(unsigned long const&, unsigned long const&, char const*) in libcef_dll_wrapper.a(cef_logging.o)
      std::string* cef::logging::MakeCheckOpString<unsigned long, unsigned int>(unsigned long const&, unsigned int const&, char const*) in libcef_dll_wrapper.a(cef_logging.o)
      std::string* cef::logging::MakeCheckOpString<unsigned int, unsigned long>(unsigned int const&, unsigned long const&, char const*) in libcef_dll_wrapper.a(cef_logging.o)
      std::string* cef::logging::MakeCheckOpString<std::string, std::string>(std::string const&, std::string const&, char const*) in libcef_dll_wrapper.a(cef_logging.o)
  "std::_Rb_tree_decrement(std::_Rb_tree_node_base*)", referenced from:
      std::_Rb_tree_iterator<std::pair<CefStringBase<CefStringTraitsUTF16> const, CefStringBase<CefStringTraitsUTF16> > >::operator--() in libcef_dll_wrapper.a(transfer_util.o)
  "std::_Rb_tree_increment(std::_Rb_tree_node_base const*)", referenced from:
      std::_Rb_tree_const_iterator<std::pair<CefStringBase<CefStringTraitsUTF16> const, CefStringBase<CefStringTraitsUTF16> > >::operator++() in libcef_dll_wrapper.a(transfer_util.o)
  "std::__throw_length_error(char const*)", referenced from:
      std::vector<CefStringBase<CefStringTraitsUTF16>, std::allocator<CefStringBase<CefStringTraitsUTF16> > >::_M_insert_aux(__gnu_cxx::__normal_iterator<CefStringBase<CefStringTraitsUTF16>*, std::vector<CefStringBase<CefStringTraitsUTF16>, std::allocator<CefStringBase<CefStringTraitsUTF16> > > >, CefStringBase<CefStringTraitsUTF16> const&) in libcef_dll_wrapper.a(transfer_util.o)
      std::vector<CefRefPtr<CefV8Value>, std::allocator<CefRefPtr<CefV8Value> > >::_M_insert_aux(__gnu_cxx::__normal_iterator<CefRefPtr<CefV8Value>*, std::vector<CefRefPtr<CefV8Value>, std::allocator<CefRefPtr<CefV8Value> > > >, CefRefPtr<CefV8Value> const&) in libcef_dll_wrapper.a(v8handler_cpptoc.o)
      std::vector<long long, std::allocator<long long> >::_M_insert_aux(__gnu_cxx::__normal_iterator<long long*, std::vector<long long, std::allocator<long long> > >, long long const&) in libcef_dll_wrapper.a(browser_ctocpp.o)
      std::vector<CefPageRange, std::allocator<CefPageRange> >::_M_insert_aux(__gnu_cxx::__normal_iterator<CefPageRange*, std::vector<CefPageRange, std::allocator<CefPageRange> > >, CefPageRange const&) in libcef_dll_wrapper.a(print_settings_ctocpp.o)
      std::vector<CefDraggableRegion, std::allocator<CefDraggableRegion> >::_M_insert_aux(__gnu_cxx::__normal_iterator<CefDraggableRegion*, std::vector<CefDraggableRegion, std::allocator<CefDraggableRegion> > >, CefDraggableRegion const&) in libcef_dll_wrapper.a(drag_handler_cpptoc.o)
      std::vector<CefRect, std::allocator<CefRect> >::_M_insert_aux(__gnu_cxx::__normal_iterator<CefRect*, std::vector<CefRect, std::allocator<CefRect> > >, CefRect const&) in libcef_dll_wrapper.a(render_handler_cpptoc.o)
      std::vector<CefRefPtr<CefPostDataElement>, std::allocator<CefRefPtr<CefPostDataElement> > >::_M_insert_aux(__gnu_cxx::__normal_iterator<CefRefPtr<CefPostDataElement>*, std::vector<CefRefPtr<CefPostDataElement>, std::allocator<CefRefPtr<CefPostDataElement> > > >, CefRefPtr<CefPostDataElement> const&) in libcef_dll_wrapper.a(post_data_ctocpp.o)
      ...
  "std::_Rb_tree_insert_and_rebalance(bool, std::_Rb_tree_node_base*, std::_Rb_tree_node_base*, std::_Rb_tree_node_base&)", referenced from:
      std::_Rb_tree<CefStringBase<CefStringTraitsUTF16>, std::pair<CefStringBase<CefStringTraitsUTF16> const, CefStringBase<CefStringTraitsUTF16> >, std::_Select1st<std::pair<CefStringBase<CefStringTraitsUTF16> const, CefStringBase<CefStringTraitsUTF16> > >, std::less<CefStringBase<CefStringTraitsUTF16> >, std::allocator<std::pair<CefStringBase<CefStringTraitsUTF16> const, CefStringBase<CefStringTraitsUTF16> > > >::_M_insert(std::_Rb_tree_node_base*, std::_Rb_tree_node_base*, std::pair<CefStringBase<CefStringTraitsUTF16> const, CefStringBase<CefStringTraitsUTF16> > const&) in libcef_dll_wrapper.a(transfer_util.o)
  "std::basic_ostream<char, std::char_traits<char> >& std::endl<char, std::char_traits<char> >(std::basic_ostream<char, std::char_traits<char> >&)", referenced from:
      cef::logging::LogMessage::~LogMessage() in libcef_dll_wrapper.a(cef_logging.o)
  "std::basic_ostream<char, std::char_traits<char> >& std::operator<<<std::char_traits<char> >(std::basic_ostream<char, std::char_traits<char> >&, char const*)", referenced from:
      CefExecuteProcess(CefMainArgs const&, CefRefPtr<CefApp>, void*) in libcef_dll_wrapper.a(libcef_dll_wrapper.o)
      CefInitialize(CefMainArgs const&, CefStructBase<CefSettingsTraits> const&, CefRefPtr<CefApp>, void*) in libcef_dll_wrapper.a(libcef_dll_wrapper.o)
      CefShutdown() in libcef_dll_wrapper.a(libcef_dll_wrapper.o)
      CefGetGeolocation(CefRefPtr<CefGetGeolocationCallback>) in libcef_dll_wrapper.a(libcef_dll_wrapper.o)
      CefAddCrossOriginWhitelistEntry(CefStringBase<CefStringTraitsUTF16> const&, CefStringBase<CefStringTraitsUTF16> const&, CefStringBase<CefStringTraitsUTF16> const&, bool) in libcef_dll_wrapper.a(libcef_dll_wrapper.o)
      CefRemoveCrossOriginWhitelistEntry(CefStringBase<CefStringTraitsUTF16> const&, CefStringBase<CefStringTraitsUTF16> const&, CefStringBase<CefStringTraitsUTF16> const&, bool) in libcef_dll_wrapper.a(libcef_dll_wrapper.o)
      CefParseURL(CefStringBase<CefStringTraitsUTF16> const&, CefStructBase<CefURLPartsTraits>&) in libcef_dll_wrapper.a(libcef_dll_wrapper.o)
      ...
  "std::basic_ostream<char, std::char_traits<char> >& std::operator<<<char, std::char_traits<char>, std::allocator<char> >(std::basic_ostream<char, std::char_traits<char> >&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)", referenced from:
      cef::logging::LogMessage::LogMessage(char const*, int, std::string*) in libcef_dll_wrapper.a(cef_logging.o)
      cef::logging::LogMessage::LogMessage(char const*, int, int, std::string*) in libcef_dll_wrapper.a(cef_logging.o)
      cef::logging::ErrnoLogMessage::~ErrnoLogMessage() in libcef_dll_wrapper.a(cef_logging.o)
      std::string* cef::logging::MakeCheckOpString<std::string, std::string>(std::string const&, std::string const&, char const*) in libcef_dll_wrapper.a(cef_logging.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [plugins/obs-browser/CEF Helper.app/Contents/MacOS/CEF Helper] Error 1
make[1]: *** [plugins/obs-browser/CMakeFiles/cef-bootstrap.dir/all] Error 2
make: *** [all] Error 2
```

--- a/README.md
+++ b/README.md
@@ -39,10 +39,10 @@ cmake -D CMAKE_XCODE_ATTRIBUTE_CLANG_CXX_LIBRARY=libc++ -G Xcode ..
 open cef.xcodeproj/
 ```
 
-TODO: tell user to move stuff, or FindCEF.cmake
-
 #### Building
 Build in Xcode (⌘+B)
+
+TODO: tell user to move stuff, or update FindCEF.cmake
 
 ### Building OBS and obs-browser
 #### Installing Dependencies
@@ -64,3 +64,38 @@ open obs-studio.xcodeproj/
 
 #### Building
 Build in Xcode (⌘+B)
+
+## Building on Windows
+
+### Building CEF
+#### Getting
+*  Download CEF Windows 64bit from [https://cefbuilds.com/](https://cefbuilds.com/)
+  *  At the time of writing this I used build 2704
+*  Extract and cd into the folder
+
+#### Setting Up Project
+* Run cmake-gui
+  * In "where is the source code", enter in the repo directory (example: C:/Users/User/Desktop/cef_binary_3.2743.1445.gdad7c0a_windows64).
+  * In "where to build the binaries", enter the repo directory path with the 'build' subdirectory (example: C:/Users/User/Desktop/cef_binary_3.2743.1445.gdad7c0a_windows64/build).
+* Press 'Configure' and select the generator that fits to your installed VS Version:
+Visual Studio 12 2013 Win64 or Visual Studio 14 2015 Win64
+* Press 'Generate' to generate Visual Studio project files in the 'build' subdirectory.
+* Open cef.sln from the 'build' subdirectory
+
+#### Building
+Build in Visual Studio
+
+TODO: tell user to move stuff, or update FindCEF.cmake
+
+### Building OBS and obs-browser
+#### Follow the OBS build instructions
+[https://github.com/jp9000/obs-studio/wiki/Install-Instructions#windows](https://github.com/jp9000/obs-studio/wiki/Install-Instructions#windows)
+
+#### Setting Up Project
+* Add add_subdirectory(obs-browser) to ./plugins/CMakeLists.txt
+* Set the CEF_ROOT_DIR path in cmake-gui for obs-studio (example: C:/Users/User/Desktop/cef_binary_3.2743.1445.gdad7c0a_windows64)
+* * Press 'Generate' to generate Visual Studio project files in the 'build' subdirectory.
+* Open obs-studio.sln from the 'build' subdirectory
+
+#### Building
+Build in Visual Studio

--- a/README.md
+++ b/README.md
@@ -23,3 +23,44 @@ window.obsstudio.onVisibilityChange(visiblity) {
 	
 };
 ```
+
+## Building on OSX
+
+### Building CEF
+#### Getting
+*  Download CEF Mac 64 from [https://cefbuilds.com/](https://cefbuilds.com/)
+  *  At the time of writing this I used build 2704
+*  Extract and cd into the folder
+
+#### Setting Up Project
+```
+mkdir build
+cmake -D CMAKE_XCODE_ATTRIBUTE_CLANG_CXX_LIBRARY=libc++ -G Xcode ..
+open cef.xcodeproj/
+```
+
+TODO: tell user to move stuff, or FindCEF.cmake
+
+#### Building
+Build in Xcode (⌘+B)
+
+### Building OBS and obs-browser
+#### Installing Dependencies
+```
+brew install ffmpeg x264 qt5 cmake
+```
+
+#### Setting Up Project
+```
+git clone --recursive https://github.com/jp9000/obs-studio.git
+cd ./obs-studio
+git clone git@github.com:kc5nra/obs-browser.git ./plugins/obs-browser
+echo "add_subdirectory(obs-browser)" >> ./plugins/CMakeLists.txt
+mkdir build
+cd ./build
+cmake -D CMAKE_PREFIX_PATH=/usr/local/opt/qt5/lib/cmake -D CEF_ROOT_DIR=/Users/username/Development/cef_binary_3.2704.1434.gec3e9ed_macosx64 -G Xcode ..
+open obs-studio.xcodeproj/
+```
+
+#### Building
+Build in Xcode (⌘+B)

